### ISSUE TITLE
False positive: structure keys reported as variables without scope prefix

### ISSUE THREAD

**Issue by mpaluchowski** [cflint/CFLint#197]
The following code code will trigger a false positive:

```
var testStructure = {
    testKey = "testValue"
};
```

reporting that `testKey` is "not declared within a var statement", where it's a structure key and hence doesn't need `var`.

**Comment by ryaneberly** [comment]
What version are you using?  
  
The latest in the dev branch, and release 0.7.3 don't flag for me. 

Full example

     component {

	public void function function1() {
		var testStructure = {
    		testKey = "testValue"
		};
	}
     }

**Comment by mjclemente** [comment]
I'm seeing the same issue with the latest master version and the latest dev version that built successfully. When prepping parameters for `QueryExecute()`, I get an error warning for each param, when it's structured this way:
```
    var params = {
      claimId = { value = claimDetail.getID(), cfsqltype = "CF_SQL_INTEGER" },
      serialNo = { value = claimDetail.getSerialNo(), cfsqltype = "CF_SQL_VARCHAR" }
    };
```

I.e.: "Variable claimId is not declared with a var statement. Use var or the local scope, or otherwise clarify the scope".

Happy to provide more information about version used, etc.

**Comment by ryaneberly** [comment]
I didn't change any related code that I'm aware of, but it works for me on latest dev build

See:
/CFLint/src/test/resources/com/cflint/tests/VarScoper/structure_keys_197.cfc
/CFLint/src/test/resources/com/cflint/tests/VarScoper/structure_keys_197.expected.txt

If it's still broken for you (on dev branch), please provide a full source example.

**Comment by mjclemente** [comment]
I can build off the latest dev branch, but when I run cflint on a folder, I get:

```
Exception in thread "main" java.lang.NullPointerException
	at com.cflint.main.CFLintMain.mergeConfigFileInFilter(CFLintMain.java:439)
	at com.cflint.main.CFLintMain.execute(CFLintMain.java:364)
	at com.cflint.main.CFLintMain.main(CFLintMain.java:252)
```

Is there something else that I should be doing? Sorry for the bother and thanks for the help.

**Comment by ryaneberly** [comment]
ok, sorry about that.   I null proofed that line - though I couldn't duplicate it with

`java -jar target\CFLint-0.8.0-all.jar -folder c:\temp`

try again?

**Comment by mjclemente** [comment]
Thanks Ryan! It builds and runs without issue now. I put together a repo with my failing example. It appears that it's actually related to a parsing error, which in turn generates the false positives.

[https://github.com/mjclemente/cflintParamStruct](url)

**Comment by ryaneberly** [comment]
Ah a multiline quoted string.  Interesting.  Any idea with dialects and versions support that?

**Comment by mjclemente** [comment]
I actually don't think it's the multiline string that's throwing it off. I think it's the boolean `!len( thing.getName() )` in the param:
```
name = { "value" = thing.getName(), "cfsqltype" = "CF_SQL_VARCHAR", null = !len( thing.getName() ) }
```

When I set it to be simply true or false, there are no issues with the parsing/validation.

And actually, I just tested including the pound signs for it, and that also works. That is, this syntax evaluates without issue: `#!len( thing.getName() )#`

I don't want to make more work for you. Including the pound signs is an acceptable workaround for my issue.

**Comment by ryaneberly** [comment]
fixed in cfparser
https://github.com/cfparser/cfparser/issues/60

it will trickle out when the next cfparser and cflint releases happen.  Until then you can build it yourself.

Thanks for taking time to identify the issue.

**Comment by mjclemente** [comment]
@ryaneberly Thanks for all the work you put into cflint!
